File: Python_version/queue/queue.py


class student():
    def __init__(self):
        self.name = ''
        self.score = 0
        self.next = None

class queue():
    def __init__(self):
        self.front = None
        self.rear = None
          
    def enqueue(self,name,score):
        newnode = student()
        newnode.name = name
        newnode.score = score
        if self.rear == None:
            self.front = newnode
        else:
            self.rear.next = newnode
        self.rear = newnode
    
    def dequeue(self):
        if self.front == None:
            print('the queue is empty')
        else:
            print('name:%s and score:%d was taken out'%(self.front.name, self.front.score))
            self.front = self.front.next
        
    def quprint(self):
        ptr = self.front
        if ptr == None:
            print('the queue is empty')
        else:
            while ptr != None:
                print('name:%s and score:%d'%(ptr.name,ptr.score))
                ptr = ptr.next

if __name__ == '__main__':
    q = queue()
    q.enqueue('howard',80)
    q.enqueue('koko',90)
    q.enqueue('dodo',100)
    q.dequeue()
    print('\nfinal output is below')
    print('---------------------------------')
    q.quprint()

File: C_version/queue/queue.c
#include <stdio.h>
#include <stdlib.h>


// create the queue 

typedef struct student{
    int score;
    struct student *next;
}node;


typedef struct queue{
    struct student *front;
    struct student *rear;
}queue;

void enqueue(queue *q, int score){
    // declaim memory
    node *newnode = (node*)malloc(sizeof(node));
    newnode -> score = score;
    newnode -> next = NULL;
    if (q -> rear == NULL){
        q-> front = newnode;
    }
    else{
        q -> rear -> next = newnode;

    }
    q -> rear = newnode;
}

void dequeue(queue *q){
    if (q -> front == NULL){
        printf("the queue is empty\n");
    }else{
        printf("the %d was taken out\n",q->front-> score);
        q-> front = q->front ->next;
    }
}

void quprintf(queue *q){
    node *ptr = q-> front;
    if (ptr == NULL){
        printf("the queue is empty");
    }
    else {
        while(ptr != NULL){
        printf("the score is %d\n", ptr->score);
        ptr = ptr->next;}
    }
}


// 請看補充資料
int main(){
    queue q;
    q.front = NULL;
    q.rear = NULL;
    
    enqueue(&q, 80);
    enqueue(&q, 90);
    enqueue(&q, 70);
    enqueue(&q, 100);
    dequeue(&q);

    printf("\nfinal output is below\n");
    printf("---------------------------------\n");
    quprintf(&q);

    return 0;
}